src/main/cpp/controllers/DrivetrainController.cpp

```diff
@@ -73,6 +73,16 @@ void DrivetrainController::AbortTrajectories() {
     m_trajectory = frc::Trajectory{};
 }
 
+bool DrivetrainController::AtHeading() const {
+    units::radian_t ref{m_r(State::kHeading)};
+    return (ref >= m_headingGoal + 0.02_rad) &&
+           (ref <= m_headingGoal - 0.02_rad);
```

Comment:
You should also be checking if the references were actually reached. That's what m_atReferences does in AtGoal(). Also, this returns true if ref is 0.02 away from m_headingGoal, which is the opposite of your intent.